akka-camel/src/main/scala/akka/camel/Producer.scala

```diff
@@ -128,7 +128,7 @@ trait ProducerSupport extends Actor with CamelSupport {
     protected def produce(endpoint: Endpoint, processor: SendProcessor, msg: Any, pattern: ExchangePattern): Unit = {
       // Need copies of sender reference here since the callback could be done
       // later by another thread.
-      val producer = self
+      val producer = context.parent
```

Comment:
I have added config for making it easier to use another dispatcher for the producer children, https://github.com/akka/akka/pull/23139